metrics/src/main/java/tech/pegasys/pantheon/metrics/Counter.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.metrics;
+
+@FunctionalInterface
+public interface Counter {
+  void inc();
```

Comment:
Do we also want an inc(int) and maybe inc(long) options?  If we gauge stuff like cumulative gas calcualated by the EVM or other such non-singleton measures a loop of inc will get tedious..

Reply:
Yes they would make sense to add, but I don't tend to add methods until they'll actually be used.   This is shaping up as an API that's designed to be public though so I'll go ahead and add them now.